### DISCUSSION TITLE
Can you use "generateFilename" async?

### DISCUSSION THREAD

**Discussion by ahamill85** [website-scraper/node-website-scraper#503] [General]
Whenever I try to use "generateFilename" asynchronously I don't get any output. The documentation has an example with the action begin async, so I thought maybe it could be.

**Comment by s0ph1e** [comment]
Hi @ahamill85 👋
Yes, async should work fine for generateFilename. Do you have an example which doesn't work for you?

**Reply by ahamill85** [reply]
I don't see any logs and there is no file output. I'm not super knowledgeable when it comes to promises, so it's entirely possible I'm screwing up my async/await functionality. Btw, I'm trying to use this package to detect a file's type and then update the filename extension to use the filetype. Reason for this, is the site I'm scraping has a lot of references/files that are without extension.

```
registerAction('generateFilename', async ({ resource }) => {
    let filename = generateFilename(resource, { defaultFilename })
    typeof resource.type === "undefined" && console.log(filename)
    const fileType = await fileTypeFromStream(resource.body)
    console.log(fileType)
    return { filename }
});
